### DISCUSSION TITLE
Welcome to merge-dracula-theme Discussions!

### DISCUSSION THREAD

**Discussion by facelessuser** [facelessuser/merge-dracula-theme#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bitsper2nd** [comment]
First of all, thanks for porting the Dracula Theme to Sublime Merge and rebuilding the color schemes(JSON format) to Sublime Text. I saw you added the Pro Variants of Dracula Theme to Sublime Merge. I was able to recreate the Color Schemes for Sublime Text by following your tips and using a color picker tool for the HEX/HSL you did not list. I have but one question. Is Dracula Pro Alucard color scheme the same as the regular Alucard you have listed in the respective repo?

**Reply by facelessuser** [reply]
Alucard is a deviation from Dracula "Classic" proper, in both cases.

Alucard was originally just the Pro themes with the new Pro palette but keeping the "Classic" background and comments. So in every respect, it was just Classic Dracula with the new palette colors for red, orange, yellow, green, cyan, purple, etc. The one difference though is that line highlights and selctions copied the approach from the Pro color schemes which use a more subtle line highlight that blends better with the background. I personally do not like Dracula proper's line highlight choice, so I prefer Alucard.

The non-Pro Alucard is the same as Pro, but it doesn't use the updated pro colors, but things like the deviation in relation to line highlight and selections carry over to it. The diff is very small between the non-Pro Alucard and the non-Pro Dracula:

```diff
--- /Users/facelessuser/Library/Application Support/Sublime Text/Packages/Color Scheme - Dracula/Dracula.sublime-color-scheme	Wed Mar 10 16:44:10 2021
+++ /Users/facelessuser/Library/Application Support/Sublime Text/Packages/Color Scheme - Dracula/Dracula (Alucard).sublime-color-scheme	Sat Jan 23 23:37:58 2021
@@ -1,5 +1,5 @@
 {
-    "name": "Dracula",
+    "name": "Dracula (Alucard)",
     "author": "Isaac Muse",
     "comments": "Derivation of: Mariana (Sublime HQ Pty Ltd, Dmitri Voronianski) using Dracula color palette (Zeno Rocha)",
     "variables":
@@ -16,10 +16,10 @@
         "purple": "hsl(265, 89%, 78%)",    // #bd93f9
 
         // Variant specific colors (change these for variants)
-        "background": "hsl(231, 15%, 18%)",        // #282a36
-        "selection": "hsl(232, 14%, 31%)",         // #44475a
-        "comment": "color(var(blue) s(27%) l(51%))",  // #6272a4
-        "line-highlight": "var(selection)",
+        "background": "hsl(231, 15%, 18%)",               // #282a36
+        "selection": "color(var(background) s(15%) l(30%))", // #414458
+        "comment": "color(var(blue) s(27%) l(51%))",         // #6272a4
+        "line-highlight": "color(var(background) s(25%) l(25%))",
         "accent": "var(blue)",
 
         // UI elements derived from color schemes
```

> Why not just update Dracula with the line highlight color and ditch the non-Pro Alucard?

I wanted to provide a "pure" Dracula for those who cared. And for those who don't, they can use Alucard which deviates from the proper/pure Dracula, but provides much better line highlights etc.

Originally, I did not ship Alucard (non-Pro) and only had the Pro version in my private repo (because sometimes I like the classic background and such), but people complained about classic line highlighting on the non-Pro Dracula, so I took Pro Alucard, changed the colors to classic, and shipped the color scheme along with the Merge theme.

Merge and Sublime users who do not use Pro and do not enable line highlighting in any of the places where syntax highlighting occurs will not notice any difference.

**Reply by bitsper2nd** [reply]
> The non-Pro Alucard is the same as Pro, but it doesn't use the updated pro colors, but things like the deviation in relation to line highlight and selections carry over to it. The diff is very small between the non-Pro Alucard and the non-Pro Dracula

I see. Full disclosure: I do have the OG Dracula Pro. So finding out the selection color was easier for me. Thanks for the insight! I forgot to change the line-highlight in the Pro Color Schemes from `var(selection)` to `color(var(background) s(25%) l(25%))`.

**Reply by facelessuser** [reply]
I do not have the OG Pro, but I'm sure my sleuthing is spot on. Most of the colors were available in the Dracula PRO website source, and the background colors were easily extracted from images. The only color not available was `red`, but was easily determined based on images and math. Oh, also, the `blue` color I use (which isn't technically a primary color from the Dracula palette) I derived from the overall Blue tone from Van Helsing. I just lightened it and saturated it like the other colors and use it as the overall accent for Van Helsing. I'm sure I'm spot on.

Now, are the selections and line highlights 100% spot on?  I don't really care, just that they look good, do their job, and aren't too much in the way 🙂. The public Alucard theme is basically what I use for the Pro themes

The funny thing is, I'm not against donating to the guy, even though I'm sure I know the palette, I just feel the $79 - $99 bucks he's asking for is absurd, and I don't care about his font recommendations and book. Though I can't fault him, he's making a good amount of money from it, and if people are willing to pay it, then more power to him 🤷🏻.

Anyways, I'll at least post this. All of my Pro variants basically use this. Comments and accents are based on the dominant color of the color scheme.

Pro - Purple
Buffy - Pink
Blade - Cyan
Lincoln - Yellow
Morbius - Red
Van Helsing - Blue

```js
        // Variant specific colors (change these for variants)
        "background": "some background color",
        "selection": "color(var(background) s(15%) l(30%))",
        "comment": "color(var(purple) s(25%) l(55%))",
        "line-highlight": "color(var(background) s(25%) l(25%))",
        "accent": "var(purple)",
```

**Reply by facelessuser** [reply]
Maybe I'll document this stuff more for Pro users.

**Comment by bitsper2nd** [comment]
> I just feel the $79 - $99 bucks he's asking for is absurd, and I don't care about his font recommendations and book. Though I can't fault him, he's making a good amount of money from it, and if people are willing to pay it, then more power to him

funny thing is that if you change the discount code on the Gumroad Dracula Pro listing(this will not work on the DraculaPro website listing)  to another country like **USPRO**  to **VNPRO** for example, the price is greatly reduced. 

> Anyways, I'll at least post this. All of my Pro variants basically use this. Comments and accents are based on the dominant color of the color scheme.

In my case, I am using orange in Morbius and Cyan in Van Helsing, since those are the colors I saw on Dracula Pro.

![image](https://user-images.githubusercontent.com/22452474/127689425-781eb4f9-46ba-47ba-81e3-5b3177365565.png)

**Reply by facelessuser** [reply]
Thanks for the info!

Additionally, I've created a guide to help with the creation of Pro variants: https://github.com/facelessuser/merge-dracula-theme#how-to-create-pro-variants.

**Reply by facelessuser** [reply]
> In my case, I am using orange in Morbius and Cyan in Van Helsing, since those are the colors I saw on Dracula Pro.

I think those are misleading.

**Reply by facelessuser** [reply]
Checkout my Morbius compared to theirs. Though they use `orange` as the button, it is clearly much more red. Granted their red is still a little orangish.

<img width="946" alt="Screen Shot 2021-07-30 at 11 30 55 AM" src="https://user-images.githubusercontent.com/1055125/127690386-9fce9e4b-02f7-4c6d-b8d0-f6acaa5ccedc.png">

**Reply by bitsper2nd** [reply]
> Checkout my Morbius compared to theirs. Though they use orange as the button, it is clearly much more red. Granted their red is still a little orangish.

I did the comparison. The colors are not misleading, it's that different values are used in the colors. 

![draculavh](https://user-images.githubusercontent.com/22452474/127695836-cbd80989-fdca-4637-92dc-39865b060268.png)

In the my version of the Pro color schemes, I am simply using HSL values for the background variable. These were converted from HEX value found on the original Dracula Pro. Since sharing is caring, here are the values I used:
- **Van Helsing** -  `"background": "hsl(210, 15%, 5%)",`
- **Buffy** - `"background": "hsl(289, 14%, 15%)",`
- **Blade** - `"background": "hsl(169, 14%, 15%)",`
- **Morbius** - `"background": "hsl(355, 14%, 15%)",`
- **Lincoln** -  `"background": "hsl(49, 14%, 15%)",`

![draculamb](https://user-images.githubusercontent.com/22452474/127695847-2caa38e9-a57b-4b84-86a4-7d151c93706e.png)

**Reply by facelessuser** [reply]
Maybe there was a misunderstanding on my part on what you are saying, but my background colors already match yours. If anything, this at least confirms I got it correct. The Morbius background is more red than orange is all I was saying. It is very close to red being `355` is very close to `0`.

 Maybe you are referring to the accent you use as orange? If so, that is fine, the accent can be whatever. I went with red personally.

**Reply by bitsper2nd** [reply]
You really nailed the colors on your Dracula Pro interpretation. If I weren't looking at the details or had the files to compare to, I wouldn't really notice any difference. Here is what I used for [reference](https://wormhole.app/rMb6o#YJLMLSqmdr08GLFHF9A7Rw).

**Reply by facelessuser** [reply]
I'll also say this. Using orange for comments deviates a lot from the samples, not that there is anything wrong with that as people should just do what looks good for them, the comments definitely seem to be more red in the official Dracula examples.

With all of that said, I also didn't try to exactly match the comments but went with something that seemed fairly close, triggering off the dominant color in the color tone in the palette. Though comparing my Morbius picture above, I think my comments are still a bit more orange than the example and maybe would make more sense using `background` instead of `red`, though the background didn't quite seem right either as it starts to mix a bit too much pink in.

If I had to guess, I suspect the comment color more likely doesn't have a hue that matches the background or the color `red`, but more `350`-ish

<img width="522" alt="Screen Shot 2021-07-30 at 12 57 24 PM" src="https://user-images.githubusercontent.com/1055125/127699896-9388851c-8455-4ad6-bf65-0e20c319829f.png">
<img width="825" alt="Screen Shot 2021-07-30 at 1 01 28 PM" src="https://user-images.githubusercontent.com/1055125/127699956-44d45ecd-983d-4723-9182-dd9b11cf9b8b.png">

But I went with something generic, so I used red 🤷🏻. Not exact, but not extremely far off, but definitely leans more to the orange than the pink where the official example leans more towards the pink.

<img width="946" alt="Screen Shot 2021-07-30 at 11 30 55 AM" src="https://user-images.githubusercontent.com/1055125/127700466-f1c2c1bd-4320-485a-a304-2ac219b879e3.png">

**Reply by facelessuser** [reply]
I just checked your reference, man I was spot on with that `350` guess 🙂.

**Reply by facelessuser** [reply]
Looking at them, it seemed comments were generally close to the dominant primary color. In those cases, so was the background. But in a couple of variants, the background was did not match closely to a primary color in the palette, in those cases, the comments follow the background pretty close. Lincoln and Morbius being perfect examples. Van Helsing, I guess, also an example as the dominant hue was not actually in the palette.

It seems like using the background to trigger the comment is probably a better choice.

**Reply by facelessuser** [reply]
I've updated comment suggestions to use the background as the base. I think in general, it gets very close for all the color schemes. Not exact, but keeps things kind of general. TBH, that is good enough for me, but people with the exact colors can just put the actual value into it.
